test/test.js

```diff
@@ -3003,3 +3003,48 @@ describe('paginate order with id_gt{,e}', () => {
 		expect(result.messages[1].id).to.be.equal(user + (4).toString());
 	});
 });
+
+describe('warm up', () => {
+	let channel;
+	let client;
+	const user = uuidv4();
+	it('shouldReuseConnection', async () => {
+		const baseUrl = 'https://chat-us-east-1.stream-io-api.com';
+		const client = getTestClient(true);
+		client.setBaseURL(baseUrl);
+		const health = await client.setUser({ id: user }, createUserToken(user));
+		client.health = health;
+		channel = await client.channel('messaging', uuidv4());
+
+		// populate cache
+		await channel.query();
+
+		// first client uses warmUp
+		const warmUpClient = getTestClientWithWarmUp();
+		warmUpClient.setBaseURL(baseUrl);
+		warmUpClient.health = await warmUpClient.setUser(
+			{ id: user },
+			createUserToken(user),
+		);
+
+		let t0 = new Date().getTime();
```

Comment:
using performance would be better for more accurate numbers since uses counters for relative time than real time system

https://nodejs.org/api/perf_hooks.html